src/Agenda.js

```diff
@@ -10,75 +10,25 @@ import { navigate } from './utils/constants'
 import { inRange } from './utils/eventLevels'
 import { isSelected } from './utils/selection'
 
-class Agenda extends React.Component {
-  constructor(props) {
-    super(props)
-    this.headerRef = React.createRef()
-    this.dateColRef = React.createRef()
-    this.timeColRef = React.createRef()
-    this.contentRef = React.createRef()
-    this.tbodyRef = React.createRef()
-  }
-
-  componentDidMount() {
-    this._adjustHeader()
-  }
-
-  componentDidUpdate() {
-    this._adjustHeader()
-  }
-
-  render() {
-    let { length, date, events, accessors, localizer } = this.props
-    let { messages } = localizer
-    let end = dates.add(date, length, 'day')
+function Agenda(props) {
+  const headerRef = useRef(null)
+  const dateColRef = useRef(null)
+  const timeColRef = useRef(null)
+  const contentRef = useRef(null)
+  const tbodyRef = useRef(null)
 
-    let range = dates.range(date, end, 'day')
+  useEffect(() => {
+    _adjustHeader()
+  })
 
-    events = events.filter(event => inRange(event, date, end, accessors))
-
-    events.sort((a, b) => +accessors.start(a) - +accessors.start(b))
-
-    return (
-      <div className="rbc-agenda-view">
-        {events.length !== 0 ? (
-          <React.Fragment>
-            <table ref={this.headerRef} className="rbc-agenda-table">
-              <thead>
-                <tr>
-                  <th className="rbc-header" ref={this.dateColRef}>
-                    {messages.date}
-                  </th>
-                  <th className="rbc-header" ref={this.timeColRef}>
-                    {messages.time}
-                  </th>
-                  <th className="rbc-header">{messages.event}</th>
-                </tr>
-              </thead>
-            </table>
-            <div className="rbc-agenda-content" ref={this.contentRef}>
-              <table className="rbc-agenda-table">
-                <tbody ref={this.tbodyRef}>
-                  {range.map((day, idx) => this.renderDay(day, events, idx))}
-                </tbody>
-              </table>
-            </div>
-          </React.Fragment>
-        ) : (
-          <span className="rbc-agenda-empty">{messages.noEventsInRange}</span>
-        )}
-      </div>
-    )
-  }
-
-  renderDay = (day, events, dayKey) => {
+  const renderDay = (day, events, dayKey) => {
     let {
       selected,
       getters,
       accessors,
       localizer,
       components: { event: Event, date: AgendaDate },
-    } = this.props
+    } = props
```

Comment:
props should be spread out at the top level (the component function body), so each function doesn't have to respread out the ones they need

Reply:
great